codegen/scripts/add-version-file.js

```diff
@@ -0,0 +1,14 @@
+import fs from 'fs';
+import { fileURLToPath } from 'url';
+import path from 'path';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const pkgPath = path.join(__dirname, '../package.json');
+const pkgContents = fs.readFileSync(pkgPath, 'utf8');
+const pkg = JSON.parse(pkgContents);
+
+const contents = `export const CODEGEN_VERSION = '${pkg.version}';`;
```

Comment:
We have to inject the version in compile time because we don't want to import the ``package.json`` into the bundle. And no, we are unable to require it dynamically because it's no longer possible under ES Modules.